packages/core/src/fragments/IfcGeometryTiler/index.ts

```diff
@@ -419,25 +421,22 @@ export class IfcGeometryTiler extends Component implements Disposable {
   }
 
   private async streamAssets() {
-    await this.onAssetStreamed.trigger(this._assets);
+    this.onAssetStreamed.trigger(this._assets);
     this._assets = null as any;
     this._assets = [];
   }
 
   private async streamGeometries() {
-    let buffer = this._streamSerializer.export(this._geometries) as Uint8Array;
+    const buffer = this._streamSerializer.export(this._geometries);
 
-    let data: StreamedGeometries = {};
+    const data: StreamedGeometries = {};
 
     for (const [id, { boundingBox, hasHoles }] of this._geometries) {
       data[id] = { boundingBox, hasHoles };
     }
 
     this.onGeometryStreamed.trigger({ data, buffer });
 
-    // Force memory disposal of all created items
```

Comment:
this is misleading. assigning `null` does not free up memory. it merely marks the memory for garbage collection, which it would do anyway once the function has run. the function does not have any closure, so there should not be any lingering memory for these two variables